lib/models/pr.js

```diff
@@ -153,70 +153,70 @@ class PR {
     get payload() {
         return this._payload;
     }
-        
+
     set payload(payload) {
         return this._payload = payload;
     }
 
     get installation() {
         return this._installation;
     }
-    
+
     get number() {
         return this._number;
     }
-    
+
     get owner() {
         return this._owner;
     }
-    
+
     get repo() {
         return this._repo;
     }
-    
+
     get id() {
         return this._id;
     }
-    
+
     get body() {
         return this._body = this._body || (this.payload.body || "").replace(/\r\n/g, "\n").trim();
     }
-    
+
     get processor() {
         return this.config.type.toLowerCase();
     }
-    
+
     get aws_s3_bucket() {
         if (process.env.ALLOW_MULTIPLE_AWS_BUCKETS == "no") {
             return process.env.AWS_BUCKET_NAME;
         }
         return this.owner == "whatwg" ? process.env.WHATWG_AWS_BUCKET_NAME : process.env.AWS_BUCKET_NAME;
     }
-    
+
     get isMultipage() {
         return !!this.config.multipage;
     }
 
     requiresPreview() {
         return !(/<!--\s*no preview\s*-->/.test(this.body));
     }
-    
+
     touchesSrcFile() {
         let relFiles = this.relevantSrcFiles;
         return this.files.some(f => relFiles.indexOf(f.filename) > -1);
     }
-    
+
     get relevantSrcFiles() {
-        return [this.config.src_file];
+        return [this.config.src_file] + this.config.includes;
```

Comment:
Mmm. This implies that we don have to do some special casing.

Have you considered just auto-detecting for include statements in the src and doing the right thing without the editor having to specify the includes?

Reply:
I thought about it briefly, but you're currently not fetching the source at all, so we'd have to add that in order to scan it for includes, and I'm not eager to write my own partial Bikeshed parser to do that scan.

We could have the server do the scan, and resolve relative to the URL, but that again raises the question of how to write the parser. We could teach *Bikeshed* to resolve includes relative to URLs, if we give it the URL of the base file instead of a local path. @tabatkins, does that sound plausible?

Reply:
> We could teach Bikeshed to resolve includes relative to URLs, if we give it the URL of the base file instead of a local path.

That would be ideal, obviously.

@marcoscaceres, would that also work for ReSpec?

Reply:
yes, I believe that would work. cc @sidvishnoi.

Reply:
I realized that having Bikeshed and Respec fetch their own includes doesn't solve the problem this line addresses of having PR Preview know when relevant source files have changed.

If we want to fetch the files here, it wouldn't be too hard to do a recursive line-by-line grep for `^path:(.*)$` ([Bikeshed](https://tabatkins.github.io/bikeshed/#including)) and `data-include="([^"]*)"|data-include='([^']*)'` ([Respec](https://github.com/w3c/respec/wiki/data--include)) to identify the watched files.

It'll probably still be good for the processors to fetch their own files on the server, since the scan here will be an overestimate, so I think we shouldn't upload everything it finds to the processor web service.

Feel free to override me on any of that, but I'll go in that direction until I hear otherwise.

Reply:
> the problem this line addresses of having PR Preview know when relevant source files have changed.... If we want to fetch the files here, it wouldn't be too hard to do a recursive line-by-line grep for..

:+1: It would be nice if pr-preview finds these dependencies rather than users specifying `includes`.

ReSpec (and spec-generator) already fetches includes relative to base URL, so this passing of `includes` files is not relevant to ReSpec, unless I'm missing the point of this PR.

Reply:
I'm happy to *either* let Bikeshed take a base URL (and then do url-fetches of its include files, rather than local file fetches), *or* add a command that parses a document and returns all the files it might be including. (You'd then have to check if those paths actually exist, because things like local boilerplate files are included by default when they exist, without an explicit indicator.)